### PR TITLE
Docs raise visibility code libraries

### PR DESCRIPTION
**Pull request by joseph-fajen**
Following feedback from the 2nd Plutus working group meeting, this change focuses on raising the visibility of the Haddock generated documentation for the Plutus code libraries. 

This PR has two changes: 
1. Some edits were made to the index.md file to put more emphasis on the Haddock docs. 
2. A new page, "Haddock documentation," was added as the first page under the Reference section.

### CODE REVIEW

File: docusaurus/docs/index.md [hunk 1]
```diff
@@ -36,8 +36,9 @@ This involves using Plutus Tx to write scripts, requiring some knowledge of the
 This guide is also meant for certification companies, certification auditors, and people who need an accurate specification. 
 See, for example:
 
-- the [Cardano Ledger Specification](https://github.com/IntersectMBO/cardano-ledger#cardano-ledger) and
-- the [Plutus Core Specification](https://github.com/IntersectMBO/plutus#specifications-and-design).
+- the [Cardano ledger specification](https://github.com/IntersectMBO/cardano-ledger#cardano-ledger),
+- the [Plutus Core specification](https://github.com/IntersectMBO/plutus#specifications-and-design), and
+- the [public Plutus code libraries](https://intersectmbo.github.io/plutus/master/) generated using Haddock.
```
Comment (olgahryniuk) [inline docusaurus/docs/index.md:41]:
```suggestion
- the [Cardano ledger specification](https://github.com/IntersectMBO/cardano-ledger#cardano-ledger)
- the [Plutus Core specification](https://github.com/IntersectMBO/plutus#specifications-and-design)
- the [public Plutus code libraries](https://intersectmbo.github.io/plutus/master/) generated using Haddock. 
```

File: docusaurus/docs/reference/haddock-documentation.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+The documentation generated by Haddock provides a comprehehsive reference for the [public Plutus code libraries](https://intersectmbo.github.io/plutus/master/), an essential resource for developers working with Haskell and Plutus Core. 
+
+### Highlighted modules
+
+Highlighted modules in the documentation include the following:
+- [PlutusTx](https://intersectmbo.github.io/plutus/master/plutus-tx/html/PlutusTx.html): Compiling Haskell to PLC (Plutus Core; on-chain code).
+- [PlutusTx.Prelude](https://intersectmbo.github.io/plutus/master/plutus-tx/html/PlutusTx-Prelude.html): Haskell prelude replacement compatible with PLC.
+- [PlutusCore](https://intersectmbo.github.io/plutus/master/plutus-core/html/PlutusCore.html): Programming language in which scripts on the Cardano blockchain are written.
+- [UntypedPlutusCore](https://intersectmbo.github.io/plutus/master/plutus-core/html/UntypedPlutusCore.html): On-chain Plutus code.
```
Comment (olgahryniuk) [inline docusaurus/docs/reference/haddock-documentation.md:17]:
```suggestion
Highlighted modules in the documentation include the following:
- [PlutusTx](https://intersectmbo.github.io/plutus/master/plutus-tx/html/PlutusTx.html): compiling Haskell to PLC (Plutus Core; on-chain code)
- [PlutusTx.Prelude](https://intersectmbo.github.io/plutus/master/plutus-tx/html/PlutusTx-Prelude.html): Haskell prelude replacement compatible with PLC
- [PlutusCore](https://intersectmbo.github.io/plutus/master/plutus-core/html/PlutusCore.html): programming language in which scripts on the Cardano blockchain are written
- [UntypedPlutusCore](https://intersectmbo.github.io/plutus/master/plutus-core/html/UntypedPlutusCore.html): on-chain Plutus code.
```

**Review by olgahryniuk — APPROVED** [review]
Minor style guide (formatting) edits

**Review by shlevy — APPROVED** [review]
Great, thanks!

**Pull request merged**